test/e2e/storage/drivers/csi.go

```diff
@@ -288,14 +288,18 @@ func (m *mockCSIDriver) PrepareTest(f *framework.Framework) (*testsuites.PerTest
 	}
 
 	o := utils.PatchCSIOptions{
-		OldDriverName:                 "csi-mock",
-		NewDriverName:                 "csi-mock-" + f.UniqueName,
-		DriverContainerName:           "mock",
-		DriverContainerArguments:      containerArgs,
-		ProvisionerContainerName:      "csi-provisioner",
-		ClusterRegistrarContainerName: "csi-cluster-driver-registrar",
-		NodeName:                      config.ClientNodeName,
-		PodInfo:                       m.podInfo,
+		OldDriverName:            "csi-mock",
+		NewDriverName:            "csi-mock-" + f.UniqueName,
+		DriverContainerName:      "mock",
+		DriverContainerArguments: containerArgs,
+		ProvisionerContainerName: "csi-provisioner",
+		NodeName:                 config.ClientNodeName,
+		PodInfo:                  m.podInfo,
+		CanAttach:                &m.attachable,
+		VolumeModes: []storagev1beta1.CSIVolumeMode{
```

Comment:
Should the field name be DriverMode?  VolumeMode is generally used to describe filesystem vs raw block

Reply:
There is no such concept as a "driver mode" anymore. That was a limitation of the alpha implementation in 1.15.

We've tried to to come up with a good name, but so far the best we came up with is "[CSI]VolumeModes". I left out the "CSI" here for the sake of brevity. In the definition here, the CSIVolumeMode type clarifies which "mode" is meant, and in yaml it also reads well:

```
VolumeModes:
- Persistent
- Ephemeral
```